src/guacd/conf-args.c

```diff
@@ -122,3 +128,15 @@ int guacd_conf_parse_args(guacd_config* config, int argc, char** argv) {
 
 }
 
+const char* guacd_conf_get_path(int argc, char** argv) {
+
+    /* Find -c option */
+    int i;
+    for(i=1; i<argc-1; i++) {
+        if(strcmp("-c", argv[i]) == 0) return argv[i+1];
```

Comment:
Manually re-parsing the command line arguments seems a poor approach. Why not parse within `guacd_conf_parse_args()` fully leveraging `getopt()`? Couldn't this setting live on the `guacd_config` struct like the others?

Reply:
> Why not parse within `guacd_conf_parse_args()` fully leveraging `getopt()`?

We need to find out what is the path to the configuration file before we parse command line arguments, since we have to apply any settings from config file before we apply any settings from command line (the latter overrides the former).

Running `while(getopt(...)...)` may cause permutation of arguments, that's why I've decided to manually parse arguments list.

> Couldn't this setting live on the `guacd_config` struct like the others?

I believe that `guacd_config` struct should store contents of a guacd configuration file so storing path to the configuration file is not needed inside it. We use it only once - when loading configuration file.

Reply:
I really don't like the idea of ignoring `-c` initially and manually re-parsing the arguments later. There has to be a better way.

What about parsing any file specified with `-c` immediately upon reading that `-c`? Arguments parsed before the file would be overridden by that file, while arguments which follow would override further. This would also open up the possibility of reading multiple files through specifying `-c` multiple times.

Reply:
> I really don't like the idea of ignoring `-c` initially and manually re-parsing the arguments later. There has to be a better way.

More precisely, it's the other way round: at first we manually search for `-c` (then load config) and then we parse all the arguments. Nevertheless, I'm not happy with this solution either, but it seems that _getopt()_ is not helping.

> What about parsing any file specified with `-c` immediately upon reading that `-c`?

That's a nice solution! I'm not sure whether it's quite common behaviour, but we can try it. :+1: